realm-core/lib/realm/persistence.rb

```diff
@@ -20,22 +20,13 @@ def initialize(repo, msg: "Cannot write using read-only repository #{repo.class}
       end
     end
 
-    extend Dry::Initializer
-
-    with_options reader: false do
-      param :root_module
-      param :container
-      option :type
-      option :url
-      option :class_path
-      option :migration_path
-      option :repositories
+    def self.setup(...)
```

Comment:
Well, you have to pass something that `initialize` expects otherwise it fails anyway. In this way id the initialize` will be changed you don't have to come back to `setup` method and modify it.

Reply:
I see , thanks